### DISCUSSION TITLE
Exception Handling Takes too long

### DISCUSSION THREAD

**Discussion by raqibhayder** [encode/starlette#1294] [Q&A] [answered]
I have a FastAPI application running in production which runs fine for the most part, but there are bursts where the requests take too long to execute. 

Below is the Transaction Trace of the request. What I do not understand is why the exception handling takes 19 seconds to complete. Any pointers or direction would be much appreciated.

![image](https://user-images.githubusercontent.com/1992290/134939492-ecd701f6-e476-43dc-96fe-690c95ce8ef9.png)

--
I have cross posted this issue in FastAPI discussion as well. [source](https://github.com/tiangolo/fastapi/discussions/3951)

**Comment by Kludex** [comment, marked as answer]
It shouldn't take this long.

I'll close this as outdated.

**Reply by rushilsrivastava** [reply]
this is marked as outdated, but I think this is still happening. I will work on a benchmark

**Comment by js-d-coder** [comment]
I am still facing this issue when using starlette v0.19.1. Please check this screenshot for details:
<img width="1681" alt="Screenshot 2025-05-06 at 8 05 27 PM" src="https://github.com/user-attachments/assets/c5e63b77-f9fd-41aa-9485-2eaeda8f3bcd" />

Notice, it takes many seconds to handle the response.

**Reply by Kludex** [reply]
We are in Starlette 0.46.2, and 0.19.1 is from April 2022... Please bump it.
